service/worker/indexer/esProcessor.go

```diff
@@ -112,7 +112,8 @@ func (p *esProcessorImpl) Stop() {
 // Add an ES request, and an map item for kafka message
 func (p *esProcessorImpl) Add(request elastic.BulkableRequest, key string, kafkaMsg messaging.Message) {
 	if p.mapToKafkaMsg.Contains(key) { // handle duplicate delete message
-		p.ackKafkaMsg(key)
+		kafkaMsg.Ack()
```

Comment:
One thing that `ackKafkaMsg` does is it removes the key from `mapToKafkaMsg`. Is that intentionally left out?

Reply:
Yes. Say handle duplicate event e1,e2 in kafka of same key, when e2 comes and it found map contains key: Old version is ack and remove the e1, which has race condition that can cause e1 being acked twice; New version just ack e2.